src/main/java/jenkins/plugins/git/GitSCMFileSystem.java

```diff
@@ -252,19 +254,23 @@ public SCMFileSystem build(@NonNull Item owner, @NonNull SCM scm, @CheckForNull
                     git.using(tool.getGitExe());
                 }
                 GitClient client = git.getClient();
-                client.addDefaultCredentials(CredentialsMatchers.firstOrNull(
-                        CredentialsProvider.lookupCredentials(
-                                StandardUsernameCredentials.class,
-                                owner,
-                                ACL.SYSTEM,
-                                URIRequirementBuilder.fromUri(remote).build()
-                        ),
-                        CredentialsMatchers.allOf(
-                                CredentialsMatchers.withId(config.getCredentialsId()),
-                                GitClient.CREDENTIALS_MATCHER
-                        )
-                        )
-                );
+
+                StandardUsernameCredentials defaultsCreds = null;
+                String credsId = config.getCredentialsId();
+                if (credsId != null) {
```

Comment:
@MarkEWaite the thing is that IdMatcher (used by CredentialsMatcher.withId()) expects something which is not null, so I think that would cause a failure and needs to be controlled.

CredentialsMatcher.firstOrNull() was already being used, so my understanding is that not having a default credential is an allowed possibility. But sure, let's wait @stephenc to shed some light on this.